Classes/Client/YLPClient.m

```diff
@@ -35,18 +35,82 @@ - (NSURLRequest *)requestWithPath:(NSString *)path {
 }
 
 - (NSURLRequest *)requestWithPath:(NSString *)path params:(NSDictionary *)params {
-    return [TDOAuth URLRequestForPath:path
-                        GETParameters:params
-                               scheme:@"https"
-                                 host:kYLPAPIHost
-                          consumerKey:self.consumerKey
-                       consumerSecret:self.consumerSecret
-                          accessToken:self.token
-                          tokenSecret:self.tokenSecret];
+    NSURLComponents *urlComponents = [[NSURLComponents alloc] init];
+    urlComponents.scheme = @"https";
+    urlComponents.host = kYLPAPIHost;
+    urlComponents.path = path;
+
+    NSMutableArray *queryItems = [NSMutableArray array];
+    for (NSString *name in params) {
+        NSString *value = [params[name] description];
+        NSURLQueryItem *queryItem = [NSURLQueryItem queryItemWithName:name value:value];
+        [queryItems addObject:queryItem];
+    }
+    if (queryItems.count > 0) {
+        urlComponents.queryItems = queryItems;
+    }
+
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:urlComponents.URL];
+    request.HTTPMethod = @"GET";
+    NSString *authHeader = [NSString stringWithFormat:@"Bearer %@", self.accessToken];
+    [request setValue:authHeader forHTTPHeaderField:@"Authorization"];
+
+    return request;
+}
+
++ (NSCharacterSet *)URLEncodeAllowedCharacters {
```

Comment:
Feels a bit like overkill to me, where I'll have to add an `@interface` and an `@implementation` and make the method be prefixed, all for something that I don't want to expose publicly and is just used in one place here. Let me know if you feel strongly about this and I can still make it happen!